core/authentication_api.php

```diff
@@ -541,6 +541,30 @@ function auth_generate_random_password() {
 }
 
 /**
+ * Generate a random string to use when we should not store the actual password.
+ * When relying on an external authentication system (e.g. LDAP), we should not
+ * store the user's password for security reasons. To avoid creating another
+ * security hole by populating the password field with a known value, we use a
+ * random string that can be identified as a 'nocache' password by its prefix.
+ * @return string Random password, prefixed by '!'
+ * @access public
+ */
+function auth_generate_nocache_password() {
```

Comment:
Let's use auth_generate_no_login_password() -- since nocache is influenced by LDAP, but there are other scenarios where we don't want to enable logins, e.g. when a user is created and we want them to reset the password, bot accounts, etc.  Apply the same naming to the other method.